model.py

```diff
@@ -0,0 +1,1232 @@
+        self.dane["PN"].append(pn)
+        self.dane["Numer pojemnika"].append(numer_pojemnika)
+        self.dane["Ilość sztuk"].append(ilosc_sztuk)
+        self.dane["Powód sztuki NOK"].append(powod_nok)
+        self.dane["Działania"].append(dzialania)
+        self.dane["Data"].append(data)
+        self.dane["Godzina"].append(godzina)
+        tabela = [entry.get() for entry in self.table_entries]
+        self.dane["Tabela"].append(tabela)
+
+        self.generuj_raport()
+
+        self.button_add["state"] = "disabled"
+        self.button_verify["state"] = "normal"
+
+        self.entry_pn.delete(0, tk.END)
+        self.entry_container.delete(0, tk.END)
+        self.entry_quantity.delete(0, tk.END)
+        self.entry_powod_nok.delete(0, tk.END)
+
+        for entry in self.table_entries:
+            entry.delete(0, tk.END)
+
+        self.label_quantity.config(text="Ilość sztuk NOK:")
+
+        if not self.dane["PN"]:
+            messagebox.showwarning("Ostrzeżenie", "Brak danych do wygenerowania raportu!")
+            return
+
+        file_path = os.path.join(os.path.dirname(__file__), "spis.xlsx")
+        if os.path.exists(file_path):
+            existing_df = pd.read_excel(file_path)
+            self.dane_df = pd.DataFrame(self.dane)
+            combined_df = pd.concat([existing_df, self.dane_df], ignore_index=True)
+        else:
+            combined_df = pd.DataFrame(self.dane)
+
+        combined_df.to_excel(file_path, index=False)
+        messagebox.showinfo("Sukces", "Raport zaktualizowany i zapisany jako spis.xlsx")
+
+        self.slider_dzialania.set("")
+
+        self.generuj_raport()
+
+        self.button_add["state"] = "disabled"
+        self.button_verify["state"] = "normal"
+
+        self.entry_pn.delete(0, tk.END)
+        self.entry_container.delete(0, tk.END)
+        self.entry_quantity.delete(0, tk.END)
+        self.entry_powod_nok.delete(0, tk.END)
+
+        for entry in self.table_entries:
+            entry.delete(0, tk.END)
+
+        self.label_quantity.config(text="Ilość sztuk NOK:")
+
+        if not self.dane["PN"]:
+            messagebox.showwarning("Ostrzeżenie", "Brak danych do wygenerowania raportu!")
+            return
+
+        file_path = os.path.join(os.path.dirname(__file__), "spis.xlsx")
+        if os.path.exists(file_path):
+            existing_df = pd.read_excel(file_path)
+            self.dane_df = pd.DataFrame(self.dane)
+            combined_df = pd.concat([existing_df, self.dane_df], ignore_index=True)
+        else:
+            combined_df = pd.DataFrame(self.dane)
+
+        combined_df.to_excel(file_path, index=False)
+        messagebox.showinfo("Sukces", "Raport zaktualizowany i zapisany jako spis.xlsx")
+
+        self.slider_dzialania.set("")
+
+    def generuj_raport(self):
+        pass
+
+    def otworz_link(self):
+        import webbrowser
+        webbrowser.open("https://www.youtube.com/watch?v=3dHpEfmegOA")

```

Comment:
importy się robi na górze strony nie w funkcji, a zwłaszcza jak nie ma ku temu powodu. Czy jak otworzysz ten jak mniemam socket to czy on się sam zamknie jak już skończysz korzystać z tego? Doczytaj w dokumentacji

Reply:
Funkcja niepotrzebna IMO